propolis/src/hw/nvme/cmds.rs

```diff
@@ -545,14 +545,15 @@ enum PrpNext {
 ///     - 2 because PRP entries are expected to be 32-bit aligned.
 ///
 /// See NVMe 1.0e Section 4.3 Physical Region Page Entry and List
-const PRP_LIST_MAX: u16 = 511;
+const PRP_LIST_MAX: u16 = 511; // XXX: 512?
```

Comment:
Hmm, as written you're right it is a little confusing. This is more accurately the last valid PRP list entry index (which start at 0). Maybe it'd be better to set this to 512 then just `- 1` as necessary below.

Reply:
I clarified the comment in eb141578c3763d1e1ed035fe307159b20ae8e5fe